docs/querying/functions.md

```diff
@@ -464,6 +464,89 @@ by the number of seconds under the specified time range window, and should be
 used primarily for human readability.  Use `rate` in recording rules so that
 increases are tracked consistently on a per-second basis.
 
+## `info()` (experimental)
+
+_The `info` function is introduced as an experiment where we try to improve UX
+around including labels from [info metrics](https://grafana.com/blog/2021/08/04/how-to-use-promql-joins-for-more-effective-queries-of-prometheus-metrics-at-scale/#info-metrics).
+The behavior of this function may change in future versions of Prometheus,
+including its removal from PromQL. info has to be enabled via the 
+[feature flag](../feature_flags.md#experimental-promql-functions) `--enable-feature=promql-experimental-functions`._
+
+`info(v instant-vector, [data-label-selector string])` finds, for each time 
+series in v, all info series with corresponding _identifying_ labels (more on
+this later), and adds the union of their _data_ (i.e., non-identifying) labels
+to the time series. The second, optional, argument is a collection of data
+label matchers that may be used to constrain which info series to consider
+and which data labels to pick from them.
```

Comment:
We cannot type the 2nd argument as `string` because that would imply quotes. I would write `instant-vector` and then explain the restrictions in the text. (To avoid introducing the concept of an entirely new node type at this point. I would still argue we want to have a "real" instant-vector in the final version, but as discussed elsewhere, maybe not, and we can defer that decision to later.)

Suggestion with some tweaks:

```suggestion
`info(v instant-vector, [data-label-selector instant-vector])` finds, for each time 
series in `v`, all info series with matching _identifying_ labels (more on
this later), and adds the union of their _data_ (i.e., non-identifying) labels
to the time series. The second argument `data-label-selector` is optional.
It is not a real instant vector, but uses a subset of its syntax.
It must start and end with curly braces (`{ ... }`) and may only contain label matchers.
The label matchers are used to constrain which info series to consider
and which data labels to add to `v`.
```